kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/operator/KroxyliciousOperatorBundleInstaller.java

```diff
@@ -0,0 +1,402 @@
+            this.extraLabels = new HashMap<>();
+        }
+    }
+
+    private static List<File> getOperatorFiles() {
+        if (operatorFiles == null || operatorFiles.isEmpty()) {
+            operatorFiles = Arrays.stream(Objects.requireNonNull(new File(Constants.PATH_TO_OPERATOR_INSTALL_FILES).listFiles())).sorted()
+                    .filter(File::isFile)
+                    .toList();
+        }
+        return operatorFiles;
+    }
+
+    private static List<File> getFilteredOperatorFiles(Predicate<File> predicate) {
+        return getOperatorFiles().stream().filter(predicate).toList();
+    }
+
+    /**
+     * Perform application of ServiceAccount, Roles and CRDs needed for proper cluster operator deployment.
+     * Configuration files are loaded from kroxylicious-operator directory.
+     */
+    public void applyClusterOperatorInstallFiles(String namespaceName) {
+        for (File operatorFile : getFilteredOperatorFiles(installFiles)) {
+            final String resourceType = operatorFile.getName().split("\\.")[1];
+
+            if (resourceType.equals(Constants.NAMESPACE)) {
+                Namespace namespace = ReadWriteUtils.readObjectFromYamlFilepath(operatorFile, Namespace.class);
+                if (!NamespaceUtils.isNamespaceCreated(namespace.getMetadata().getName())) {
+                    kubeClient().getClient().resource(namespace).create();
+                }
+            }
+            else {
+                try {
+                    kubeClient().getClient().load(new FileInputStream(operatorFile.getAbsolutePath()))
+                            .inNamespace(namespaceName)
+                            .create();
+                }
+                catch (FileNotFoundException e) {
+                    throw new UncheckedIOException(e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Prepare environment for cluster operator which includes creation of namespaces, custom resources and operator
+     * specific config files such as ServiceAccount, Roles and CRDs.
+     * @param clientNamespace namespace which will be created and used as default by kube client
+     * @param namespaces list of namespaces which will be created
+     */
+    public void prepareEnvForOperator(String clientNamespace, List<String> namespaces) {
+        try {
+            applyCrds();
+        }
+        catch (FileNotFoundException e) {
+            throw new UncheckedIOException(e);
+        }
+        applyClusterOperatorInstallFiles(clientNamespace);
+
+        if (cluster.cluster().isOpenshift() && kubeClient().getNamespace(Environment.KROXY_ORG) != null) {
+            for (String namespace : namespaces) {
+                LOGGER.debug("Setting group policy for Openshift registry in Namespace: {}", namespace);
+                Exec.exec(Arrays.asList("oc", "policy", "add-role-to-group", "system:image-puller", "system:serviceaccounts:" + namespace, "-n", Environment.KROXY_ORG));
```

Comment:
Isn't this required for OpenShift Image Streams?  WHere will this project actually need Image Streams? 
https://docs.openshift.com/container-platform/4.17/openshift_images/managing_images/using-image-pull-secrets.html

Reply:
removed